### PR TITLE
Unify Dashboard & Webhook events

### PR DESCRIPTION
**Pull request by bmc-msft**
This change unifies the previously adhoc SignalR events and Webhooks into a single event format.

### CODE REVIEW

File: src/cli/onefuzz/status/cache.py [hunk 1]
```diff
@@ -4,39 +4,75 @@
...
+MESSAGE = Tuple[datetime, EventType, str]
 
 MINUTES = 60
 HOURS = 60 * MINUTES
 DAYS = 24 * HOURS
 
 
-def fmt_delta(data: timedelta) -> str:
-    result = []
+class MiniNode(BaseModel):
```
Comment (ranweiler) [inline src/cli/onefuzz/status/cache.py:49]: What are these `Mini*` classes?

File: src/cli/onefuzz/status/cache.py [hunk 2]
```diff
@@ -106,194 +139,195 @@ def __init__(
...
-            elif isinstance(v, (int, str)):
-                data[k] = v
-
-        as_str = fmt(data)
-        messages = [x for x in self.messages if (x[1:] != [name, as_str])][-99:]
-        messages += [(datetime.now(), name, as_str)]
-
+        messages = [x for x in self.messages][-99:]
+        messages += [
+            (datetime.now(), message.event_type, message.event.json(exclude_none=True))
```
Comment (ranweiler) [inline src/cli/onefuzz/status/cache.py:176]: This isn't new, but should we be explicitly specifying the TZ as UTC in all our calls to `now()`?

File: src/cli/onefuzz/status/cache.py [hunk 3]
```diff
@@ -311,63 +345,60 @@ def should_render_job(self, job: Job) -> bool:
...
         return True
 
+    def job_stopped(self, event: EventJobStopped) -> None:
+        if event.job_id in self.jobs:
+            del self.jobs[event.job_id]
+
     def render_jobs(self) -> List[Tuple]:
         results: List[Tuple] = []
 
-        for (timestamp, job) in sorted(self.jobs.values(), key=lambda x: x[0]):
```
Comment (ranweiler) [inline src/cli/onefuzz/status/cache.py:317]: Why do we no longer sort by timestamp?

**Review by ranweiler — APPROVED** [review]

File: src/cli/onefuzz/status/cache.py (hunk 3 above)
Comment (bmc-msft) [reply src/cli/onefuzz/status/cache.py:317]: Once I had a 3k concurrent tasks in the view, sorting on timestamp caused the data to jump around far too frequently to be useful.

File: src/cli/onefuzz/status/cache.py (hunk 2 above)
Comment (bmc-msft) [reply src/cli/onefuzz/status/cache.py:176]: In general, yes.  However, in the case of displaying to users, it made sense to use `now()` here.

This only displays the local timestamp of when the message was received.

File: src/cli/onefuzz/status/cache.py (hunk 1 above)
Comment (bmc-msft) [reply src/cli/onefuzz/status/cache.py:50]: We don't send the full context of the Node/Task/Job, etc via SignalR.  As such, these are slimmed down representations of the models specifically for the context of `status top`.

Added a comment to each of them to describe that.

**Pull request merged**